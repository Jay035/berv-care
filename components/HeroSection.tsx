"use client";
import Image from "next/image";
import img from "../public/doctor-1.png";
import GetStartedBtn from "./GetStartedBtn";
import { useGSAP } from "@gsap/react";
import SplitType from "split-type";
import gsap from "gsap";

type Props = {};

export default function HeroSection({}: Props) {
  useGSAP(() => {
    const heading = new SplitType("#hero_title", { types: "words" });
    const paragraph = new SplitType(".hero_text", { types: "words" });

    gsap.from([heading.words, paragraph.words, "#hero_btn", "#hero_img"], {
      y: 24,
      opacity: 0,
      duration: 0.8,
      stagger: { amount: 1 },
      ease: "ease",
    });
  });
  return (
    <section className="grid items-center lg:grid-cols-2 gap-[7.25rem] mt-4 lg:h-[70vh] lg:justify-center lg:mt-0 lg:mb-0 mb-[3.375rem] px-8 sm:px-[9.5vw]">
      <div className="flex flex-col items-center lg:items-start md:w-[32rem] mx-auto lg:w-full">
        <p className="hero_text text-center lg:text-left font-semibold text-base xl:text-lg text-[#6B7280] mb-1">
          Find the Care You Deserve, With Ease
        </p>
        <h1
          id="hero_title"
          className="font-extrabold text-center lg:text-left text-[#14532D] text-3xl md:text-4xl xl:text-6xl"
        >
          Your Pathway to Trusted Care Providers
        </h1>
        <p className="hero_text text-center lg:text-left mt-4 text-lg xl:text-xl text-[#374151]">
          With <span className="font-bold text-[#14532D]">Berv-Care</span>,
          finding the nearest care has never been easier. Start your journey to
          exceptional care today.
          {/* You can also access
          the best body tests and doctors without leaving home. Need help with
          booking your test? Our quiz is here to guide you.  */}
          {/* Embrace peace of mind
          knowing that your loved ones are in the hands of dedicated caregivers
          who genuinely care.  */}
        </p>
        {/* CTAs */}
        <div id="hero_btn" className="mt-12 text-lg">
          <GetStartedBtn />
        </div>
      </div>
      <div className="">
        <Image
          id="hero_img"
          placeholder="blur"
          src={img}
          className="w-full"
          alt="doctor treating patient"
        />
      </div>
    </section>
  );
}
