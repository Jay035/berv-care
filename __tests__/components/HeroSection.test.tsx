import HeroSection from "@/components/HeroSection";
import { render, screen } from "@testing-library/react";

describe("Hero section", () => {
  it("should render hero section properly", () => {
    const { container } = render(<HeroSection />);
    const heading = screen.getByRole("heading", {
      name: /Your Pathway to Trusted Care Providers/i,
    });

    expect(heading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
