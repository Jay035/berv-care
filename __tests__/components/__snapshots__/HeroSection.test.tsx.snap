// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero section should render hero section properly 1`] = `
<div>
  <section
    class="grid items-center lg:grid-cols-2 gap-[7.25rem] mt-4 lg:h-[70vh] lg:items-center lg:justify-center lg:mt-0 lg:mb-0 mb-[3.375rem] px-8 sm:px-[9.5vw]"
  >
    <div
      class="w-full mx-auto text-center md:w-[32rem] lg:w-full lg:text-left"
    >
      <p
        class="font-semibold text-base xl:text-lg text-[#6B7280] mb-1"
      >
        Find the Care You Deserve, With Ease
      </p>
      <h1
        class="font-extrabold text-[#14532D] text-3xl md:text-4xl xl:text-6xl"
      >
        Your Pathway to Trusted Care Providers
      </h1>
      <p
        class="mt-4 text-lg xl:text-xl text-[#374151]"
      >
        With 
        <span
          class="font-bold text-[#14532D]"
        >
          Berv-Care
        </span>
        , finding the nearest care has never been easier. Start your journey to exceptional care today.
      </p>
      <div
        class="mt-12 text-lg"
      >
        <a
          class="py-4 md:py-5 w-fit px-8 text-lg md:px-14 bg-[#14532D] text-white rounded-[50px] transition hover:text-black hover:bg-white hover:border hover:border-black"
          href="/hospitals"
        >
          Get Started
        </a>
      </div>
    </div>
    <div
      class=""
    >
      <img
        alt="doctor treating patient"
        class="w-full"
        src="/doctor-1.png"
      />
    </div>
  </section>
</div>
`;
